cmd/controller/app/controller.go

```diff
@@ -177,6 +177,7 @@ func buildControllerContext(opts *options.ControllerOptions) (*controller.Contex
 			HTTP01SolverResourceRequestMemory: HTTP01SolverResourceRequestMemory,
 			HTTP01SolverResourceLimitsCPU:     HTTP01SolverResourceLimitsCPU,
 			HTTP01SolverResourceLimitsMemory:  HTTP01SolverResourceLimitsMemory,
+			DNS01CheckAuthoritative:           opts.DNS01RecursiveNameserversOnly,
```

Comment:
I'm pretty sure these two need to be inverted? IMO, it'd be simpler for us to `s/DNS01CheckAuthoritative/DNS01RecursiveOnly` and then just invert (i.e. add `!`) all current uses of `DNS01CheckAuthoritative`

Reply:
hmmmm, I think possibly just do the negate in the context setup. This way the language stays consistent lower in the code. There is a higher change that the context variable will get used in more than 1 place so the chance of someone forgetting to negate it may happen. Just my thought.

Happy to go with either option. Going to submit the change for the above but more than happy to change it. Just want to see if these tests pass.